### DISCUSSION TITLE
Hiding portions of examples in code blocks?

### DISCUSSION THREAD

**Discussion by hayesall** [maaslalani/slides#134] [Q&A] [answered]
Sometimes there's verbose setup in examples:

```go
package main

import "fmt"

func main() {
  fmt.Println("Execute code directly inside the slides")
}
```

---

But this is a little verbose.

It would be nice if slides could show:

```go
  fmt.Println("Execute code directly inside the slides")
```

```
Execute code directly inside the slides
```

---

Is there something similar to [RustDocs example hiding](https://doc.rust-lang.org/stable/rustdoc/documentation-tests.html#hiding-portions-of-the-example)?

Like annotating:

```go
# package main
# 
# import "fmt"
#
# func main() {
  fmt.Println("Execute code directly inside the slides")
# }
```

**Comment by maaslalani** [comment]
This is a fantastic idea! Thanks so much for suggesting the ability to hide verbose code.

Let me know if you are interested in implementing this, I would happily accept a PR to do this, otherwise I can tackle it and let you know once it is implemented.

**Reply by maaslalani** [reply]
Another idea I was contemplating was adding the ability to run small code snippets for Go by allowing people to write simple lines and then performing all the auto imports and everything to compile and run it. But, I think that would be quite Go specific and this annotating solution is very elegant and would work with all languages.

**Reply by hayesall** [reply]
I tried local modifications in `code.go`, but it's nowhere near ready for human consumption.

I now think it's better to mark "hidden" lines with something at the end though:

```go
package main#
import "fmt"#
#
func main() {#
  fmt.Println("Execute code directly inside the slides")#
}#
```

Rust doesn't care too much about whitespace, but the Python example I tried freaked out.

**Reply by hayesall** [reply]
If you want to convert this to an *enhancement issue*, it might be more appropriate than a Q&A discussion.

I don't have a timeline yet for when I'll be done with this though.

**Reply by maaslalani** [reply]
I like the character at the beginning since its easier to toggle it (if you're writing the code in a code editor) but I'm open to having it at the end too.

Created an issue, don't worry about the timeline :) You are the first person to ask for this but I think it's a great and useful idea to include:

https://github.com/maaslalani/slides/issues/135

**Comment by maaslalani** [comment, marked as answer]
See issue https://github.com/maaslalani/slides/issues/135
